presto-main/src/main/java/io/prestosql/execution/CreateMaterializedViewTask.java

```diff
@@ -0,0 +1,147 @@
+        accessControl.checkCanCreateTable(session.toSecurityContext(), name);
+
+        String sql = getFormattedSql(statement.getQuery(), sqlParser);
+
+        Analysis analysis = analyzeStatement(statement, session, metadata, accessControl, parameters, parameterLookup, stateMachine.getWarningCollector(), sql);
+
+        List<ViewColumn> columns = analysis.getOutputDescriptor(statement.getQuery())
+                .getVisibleFields().stream()
+                .map(field -> new ViewColumn(field.getName().get(), field.getType().getTypeId()))
+                .collect(toImmutableList());
+
+        Optional<String> owner = Optional.of(session.getUser());
+
+        CatalogName catalogName = metadata.getCatalogHandle(session, name.getCatalogName())
+                .orElseThrow(() -> new PrestoException(NOT_FOUND, "Catalog does not exist: " + name.getCatalogName()));
+
+        Map<String, Expression> sqlProperties = mapFromProperties(statement.getProperties());
+        Map<String, Object> properties = metadata.getTablePropertyManager().getProperties(
+                catalogName,
+                name.getCatalogName(),
+                sqlProperties,
+                session,
+                metadata,
+                accessControl,
+                parameterLookup);
+
+        Map<String, Object> finalProperties = combineProperties(sqlProperties.keySet(), properties);
+
+        ConnectorMaterializedViewDefinition definition = new ConnectorMaterializedViewDefinition(
+                sql,
+                null,
+                session.getCatalog(),
+                session.getSchema(),
+                columns,
+                owner,
+                false,
+                statement.getComment(),
+                finalProperties);
+
+        metadata.createMaterializedView(session, name, definition, statement.isReplace(), statement.isNotExists());
+
+        return immediateFuture(null);
+    }
+
+    private Analysis analyzeStatement(Statement statement, Session session, Metadata metadata, AccessControl accessControl, List<Expression> parameters,
+        Map<NodeRef<Parameter>, Expression> parameterLookup, WarningCollector warningCollector, String query)
+    {
+        Analyzer analyzer = new Analyzer(session, metadata, sqlParser, accessControl, Optional.empty(), parameters, parameterLookup, warningCollector);
+        return analyzer.analyze(statement);
+    }
+
+    private static Map<String, Object> combineProperties(Set<String> specifiedPropertyKeys, Map<String, Object> defaultProperties)
+    {
+        Map<String, Object> finalProperties = new HashMap<>();
+        for (Map.Entry<String, Object> entry : defaultProperties.entrySet()) {
+            if (specifiedPropertyKeys.contains(entry.getKey()) || !finalProperties.containsKey(entry.getKey())) {
+                finalProperties.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return finalProperties;
```

Comment:
This method is a no-op. The `!finalProperties.containsKey(entry.getKey())` will cause every entry from `defaultProperties` to be copied to the result map. It's equivalent to just returning the input map.

Reply:
yes, you are right. Fixed.